testplan/testing/multitest/base.py

```diff
@@ -288,20 +288,26 @@ def get_test_context(self):
         :rtype: ``list`` of ``tuple``
         """
         ctx = []
-        sorted_suites = self.cfg.test_sorter.sorted_testsuites(self.cfg.suites)
+        sorted_suites = (
+            self.cfg.test_sorter.sorted_testsuites(self.cfg.suites)
+            if hasattr(self.cfg, "test_sorter")
```

Comment:
Can multitest have a default test_sorter so that we don't need to check this?

Reply:
We cannot have a default `test_sorter` (as well as `test_filter`, `stdout_style`). This is an old issue. Because in the past we have another arguments to decide which cfg option is preferred if it is defined in both parent and child instance, but that argument is removed. So, now `TestConfig` class does not have default `test_sorter`, `test_filter` and `stdout_style` because these cfg options should be retrieved from `TestRunner`. This is reasonable or the cfg option we defined in @test_plan cannot take effect. But a small problem is that in the unit test, a single `MultiTest` instance cannot run and every time we want to write a test we have to pass a lot of arguments to it (that is why `test_assertion_order` always raise an exception). So with these `hasattr` check things become simple.

Reply:
Ok, then i think it would make better sense to add the multitest to a mockplan in test code -- we are doing this in many places.